### PR TITLE
Add ability to change endpoint before a session is started

### PR DESCRIPTION
**Pull request by aggFTW**
Magic looks like:

`%_do_not_call_change_endpoint -u admin -p password -s https://cluster.com`

### CODE REVIEW

**Comment by tc0312** [comment]
LGTM

File: serverextension/sparkmagic_ext/handlers.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import json
+from notebook.utils import url_path_join
+from notebook.base.handlers import IPythonHandler
+
+
+class ReconnectHandler(IPythonHandler):
+    def get(self):
+        path = 'Temp/Untitled3.ipynb'
```
Comment (linchan-ms) [inline serverextension/sparkmagic_ext/handlers.py]: Why hardcoded path?

Comment (aggFTW) [inline serverextension/sparkmagic_ext/handlers.py]: This file is not part of this PR.
